src/librbd/internal.cc

```diff
@@ -72,6 +73,19 @@ namespace librbd {
 
 namespace {
 
+struct C_CreateV2Image : public Context {
```

Comment:
I actually wanted to pack as much as possible into ImageCtx and handover that to CreateRequest. ImageCtx has member variables such as order, features, size, etc. which are filled during Refresh, so, I kept away from filling ImageCtx directly (via additional ctor).